### PR TITLE
Add `BigDecimal(kotlin.String)` to `ForbiddenMethodCall`

### PR DESCRIPTION
**Pull request by atulgpt**
Currently, we already have one forbidden method related to BigDecimal

Fixes https://github.com/detekt/detekt/issues/5977

### CODE REVIEW

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt [hunk 1]
```diff
@@ -65,9 +65,11 @@ class ForbiddenMethodCall(config: Config) : Rule(
         valuesWithReason(
             "kotlin.io.print" to "print does not allow you to configure the output stream. Use a logger instead.",
             "kotlin.io.println" to "println does not allow you to configure the output stream. Use a logger instead.",
-            "java.math.BigDecimal.<init>(kotlin.Double)" to "using `BigDecimal.<init>(kotlin.Double)` can result in " +
-                "unexpected float point precision behavior. Use `BigDecimal.valueOf(kotlin.Double)` or " +
-                "`BigDecimal.<init>(kotlin.String)` instead.",
+            "java.math.BigDecimal.<init>(kotlin.Double)" to "using `BigDecimal(Double)` can result in " +
+                "unexpected float point precision behavior. Use `BigDecimal.valueOf(Double)` or " +
```
Comment (schalkms) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt:69]:
```suggestion
                "unexpected floating point precision behavior. Use `BigDecimal.valueOf(Double)` or " +
```

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt [hunk 2]
```diff
@@ -65,9 +65,11 @@ class ForbiddenMethodCall(config: Config) : Rule(
...
+                "`String.toBigDecimalOrNull()` instead.",
+            "java.math.BigDecimal.<init>(kotlin.String)" to "using `BigDecimal(String)` can result in " +
```
Comment (schalkms) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt:71]:
```suggestion
            "java.math.BigDecimal.<init>(kotlin.String)" to "using `BigDecimal(String)` can result in a " +
```

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCallSpec.kt [hunk 3]
```diff
@@ -754,13 +754,27 @@ class ForbiddenMethodCallSpec(val env: KotlinCoreEnvironment) {
             }
 
             @Test
-            fun `should report nothing when using BigDecimal string constructor`() {
+            fun `should report when using BigDecimal string constructor`() {
```
Comment (schalkms) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCallSpec.kt:757]: `not` is still correct here.

```suggestion
            fun `should not report when using BigDecimal string constructor`() {
```

**Review by schalkms — COMMENTED** [review]
Cool - thanks for adding this to the default arguments!

Some minor feedback to the message from my side underneath

**Review by cortinico — APPROVED** [review]

**Review by schalkms — APPROVED** [review]

**Pull request merged**